data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java

```diff
@@ -301,7 +301,18 @@ public Future<Void> close() {
             return Future.succeededFuture();
         };
 
-        return closePromise.future().compose(v -> closeF.apply(null), v -> closeF.apply(null));
+      Future<Void> compose = closePromise.future().compose(v -> closeF.apply(null), v -> closeF.apply(null));
+
+      return compose.eventually(v -> {
+          return vertx.executeBlocking(promise -> {
+              try {
+                  tokenProvider.close();
+                  promise.complete();
+              } catch (Exception e) {
+                  promise.fail(e);
+              }
+          });
+      });
```

Comment:
Why does this need to be in a future? Can't we do it like `this.client.close();` on line 300?

Reply:
No particular reason. You are right. Rereading the javadoc, AFAIUI, it can be done the same way. Updated.